src/sprite_airshipshadow.cpp

```diff
@@ -52,6 +53,55 @@ void Sprite_AirshipShadow::RecreateShadow() {
 	GetBitmap()->Blit(0, 0, *system, Rect(128+16,32,16,16), opacity);
 }
 
+void Sprite_AirshipShadow::CreateClones() {
+	if (Game_Map::GetMapId() > 0) {
+		Game_Vehicle* airship = Game_Map::GetVehicle(Game_Vehicle::Airship);
+		if (!airship) {
+			return;
+		}
+		int x = GetX();
+		int y = GetY();
+		Rect src_rect = GetSrcRect();
+
+		bool loop_x = Game_Map::GetWidth() <= 21 && Game_Map::LoopHorizontal()
+			&& (x < src_rect.width || x >= Game_Map::GetWidth() * TILE_SIZE - src_rect.width);
+
+		bool loop_y = Game_Map::GetHeight() <= 16 && Game_Map::LoopVertical()
+			&& (y < src_rect.height || y >= Game_Map::GetHeight() * TILE_SIZE - src_rect.height);
+
+		if (loop_x) {
+			clone_x = std::shared_ptr<SpriteClone>(
+				new SpriteClone(this, x + Game_Map::GetWidth() * TILE_SIZE * (x < src_rect.width ? 1 : -1), y, GetZ()));
+		}
+
+		if (loop_y) {
+			airship->SetY(airship->GetY() + (Game_Map::GetHeight() * TILE_SIZE + 1) * (y < src_rect.height ? 1 : -1));
+
+			clone_y = std::shared_ptr<SpriteClone>(
+				new SpriteClone(this, x, y + Game_Map::GetHeight() * TILE_SIZE * (y < src_rect.height ? 1 : -1), airship->GetScreenZ()));
+
+			airship->SetY(airship->GetY() - (Game_Map::GetHeight() * TILE_SIZE + 1) * (y < src_rect.height ? 1 : -1));
+		}
+
+		if (loop_x && loop_y) {
+			airship->SetY(airship->GetY() + Game_Map::GetHeight() * TILE_SIZE * (y < src_rect.height ? 1 : -1));
+
+			clone_xy = std::shared_ptr<SpriteClone>(
+				new SpriteClone(this, x + Game_Map::GetWidth() * TILE_SIZE * (x < src_rect.width ? 1 : -1),
+					y + Game_Map::GetHeight() * TILE_SIZE * (y < src_rect.height ? 1 : -1), airship->GetScreenZ()));
+
+			airship->SetY(airship->GetY() - Game_Map::GetHeight() * TILE_SIZE * (y < src_rect.height ? 1 : -1));
+		}
+
+	}
+}
+
+void Sprite_AirshipShadow::DestroyClones() {
+	clone_x.reset();
+	clone_y.reset();
+	clone_xy.reset();
```

Comment:
This SpriteClone approach is fine but is there a way to manage the clone creation in Sprite_AirshipShadow/Sprite_Character or in Spriteset Map? When you do it in Spriteset Map you could add them to the character_sprites array which means there Update function is called.
I mean you only need to create them once when the Sprite is created and then they basicly stay until the parent sprite is removed (on map change).
The 3 clones could manage there state automatically (e.g. when the sprite is not at a corner they are just Visible(false)).